src/sagemaker/jumpstart/artifacts/model_packages.py

```diff
@@ -75,3 +75,56 @@ def _retrieve_model_package_arn(
         return regional_arn
 
     raise NotImplementedError(f"Model Package ARN not supported for scope: '{scope}'")
+
+
+def _retrieve_model_package_model_artifact_s3_uri(
+    model_id: str,
+    model_version: str,
+    region: Optional[str],
+    scope: Optional[str] = None,
+    tolerate_vulnerable_model: bool = False,
+    tolerate_deprecated_model: bool = False,
+) -> Optional[str]:
+    """Retrieves s3 artifact uri associated with model package.
+
+    Args:
+        model_id (str): JumpStart model ID of the JumpStart model for which to
+            retrieve the model package artifact.
+        model_version (str): Version of the JumpStart model for which to retrieve the
+            model package artifact.
+        region (Optional[str]): Region for which to retrieve the model package artifact.
+        scope (Optional[str]): Scope for which to retrieve the model package artifact.
+        tolerate_vulnerable_model (bool): True if vulnerable versions of model
+            specifications should be tolerated (exception not raised). If False, raises an
+            exception if the script used by this version of the model has dependencies with known
+            security vulnerabilities. (Default: False).
+        tolerate_deprecated_model (bool): True if deprecated versions of model
+            specifications should be tolerated (exception not raised). If False, raises
+            an exception if the version of the model is deprecated. (Default: False).
+
+    Returns:
+        str: the model package artifact uri to use for the model or None.
+    """
+
+    if region is None:
+        region = JUMPSTART_DEFAULT_REGION_NAME
+
+    model_specs = verify_model_region_and_return_specs(
+        model_id=model_id,
+        version=model_version,
+        scope=scope,
+        region=region,
+        tolerate_vulnerable_model=tolerate_vulnerable_model,
+        tolerate_deprecated_model=tolerate_deprecated_model,
+    )
+
+    if scope == JumpStartScriptScope.TRAINING:
+
+        if model_specs.training_model_package_artifact_uris is None:
+            return None
+
+        regional_arn = model_specs.training_model_package_artifact_uris.get(region)
```

Comment:
rename please: `model_s3_uri`

Also question: why is there a regional map in regionalized metadata?

Reply:
Although we technically support all regions, JCP will remove regions that don't match the target region. This implementation allows us to implement regionalization in JCP _after_ releasing the production metadata.